### ISSUE TITLE
nvim-lspconfig bindings not working/showing

### ISSUE THREAD

**Issue by Vanderscycle** [siduck76/NvChad#45]
@siduck76,

Sorry for pestering you over the past few days with my issues. I may just need some confirmation, my lspconfig bindings do not work on my config (the lspconfig commands do work tho if executed in the cli/nvim terminal) and they do not show using nvim-which key. 
:lspInfo reveals that lsp is indeed working and no issues on :checkHealth

```lua
  local function map(...)
      vim.api.nvim_buf_set_keymap(bufnr, ...)
  end

  local function buf_set_option(...)
      vim.api.nvim_buf_set_option(bufnr, ...)
  end

  buf_set_option("omnifunc", "v:lua.vim.lsp.omnifunc")

  -- Mappings.
  local opts = {noremap = true, silent = true, normal = true}
  map("gD", "<Cmd>lua vim.lsp.buf.declaration()<CR>", opts)
  map("gd", "<Cmd>lua vim.lsp.buf.definition()<CR>", opts)
  map("K", "<Cmd>lua vim.lsp.buf.hover()<CR>", opts)
  map("gi", "<cmd>lua vim.lsp.buf.implementation()<CR>", opts)
  map("<C-k>", "<cmd>lua vim.lsp.buf.signature_help()<CR>", opts)
  map("<space>wa", "<cmd>lua vim.lsp.buf.add_workspace_folder()<CR>", opts)
  map("<space>wr", "<cmd>lua vim.lsp.buf.remove_workspace_folder()<CR>", opts)
  map("<space>wl", "<cmd>lua print(vim.inspect(vim.lsp.buf.list_workspace_folders()))<CR>", opts)
  map("<space>D", "<cmd>lua vim.lsp.buf.type_definition()<CR>", opts)
  map("<space>rn", "<cmd>lua vim.lsp.buf.rename()<CR>", opts)
  map("gr", "<cmd>lua vim.lsp.buf.references()<CR>", opts)
  map("<space>e", "<cmd>lua vim.lsp.diagnostic.show_line_diagnostics()<CR>", opts)
  map("[d", "<cmd>lua vim.lsp.diagnostic.goto_prev()<CR>", opts)
  map("]d", "<cmd>lua vim.lsp.diagnostic.goto_next()<CR>", opts)
  map("<space>q", "<cmd>lua vim.lsp.diagnostic.set_loclist()<CR>", opts)
```
The bindings showing for me leader key (leader+x* is binded to another plugin)
![whichKey](https://user-images.githubusercontent.com/52706578/117753486-653f6e80-b1cd-11eb-9450-3bc83e187f5f.png)

tl;dr: lsp is working flawlessly on my config but none of the bindings work.

**Comment by siduck76** [comment]
@Vanderscycle I dont mind the issues mate , Keep bringing em :D !! I will test those mappings when I get back on pc.

**Comment by Vanderscycle** [comment]
No problem I'm happy of the current LSP state, minor nitpicks at this point. Last question, in lspconfig.lua you have the bindings and options set under the function on_attach, but (excuse my ignorance) I didn't see it being called anywhere in the init.lua or in the file itself.

**Comment by siduck76** [comment]
2fe776a3866b49dc6bde0b5315c777855a195838 should resolve it

**Comment by Vanderscycle** [comment]
I'll test shortly

**Comment by siduck76** [comment]
@Vanderscycle do those mappings work for you?

**Comment by siduck76** [comment]
> @Vanderscycle nvim-lsp mappings work for you?

**Comment by Vanderscycle** [comment]
yeah, and show on which-key

**Comment by Vanderscycle** [comment]
For example, I inserted two errors in this ts file. by placing your cursor over one of the two mistake and pressing [d you will go to the next one.

![example](https://user-images.githubusercontent.com/52706578/117757269-a0916b80-b1d4-11eb-9864-a30a02f21dc1.png)

**Comment by siduck76** [comment]
does the vim.lsp.buf.hover() work for you?

**Comment by siduck76** [comment]
cool! it works for me now too but some lsp diagnostics doesnt work in js files 

ex - 
![image](https://user-images.githubusercontent.com/59060246/117758202-2fd36900-b23f-11eb-8ca3-b5640c6cea7b.png)

**Comment by Vanderscycle** [comment]
I'm learning TS, in the hopes of skipping the mind-shattering production errors that JS often creates. I'm not familiar with JS but the error should still show.

What does :LspInfo says?

**Comment by siduck76** [comment]
![image](https://user-images.githubusercontent.com/59060246/117758381-935d9680-b23f-11eb-84d7-aa2b17051c2d.png)

lspDiagnostics work for some parts

![image](https://user-images.githubusercontent.com/59060246/117758424-a8d2c080-b23f-11eb-91d2-5d30dfd01b2b.png)

**Comment by Vanderscycle** [comment]
Hmmm, idk... I'll keep my eyes open for these errors. Could be a good issue to raise on the corresponding lsp repo.
